File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        long temp = (long) num;
        
        while(temp!=1) {
            temp = temp % 2 == 0 ? temp / 2 : temp * 3 + 1;
            answer++;
            
            if(answer==500) {
                return answer = -1;    
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) {
    var answer = 0;
    
    let count = 0;
    while(num !== 1 && count !== 500) {
        if(num % 2 == 0) {
            num /= 2;
        } else {
            num *= 3;
            num++;
        }
        count++;
    }
    
    return num === 1 ? count : -1;
}